lib/facilities/local_store.rb

```diff
@@ -0,0 +1,109 @@
+# frozen_string_literal: true
+require 'facilities/bulk_client'
+
+module Facilities
+  class LocalStore
+    attr_reader :adapter
+
+    GIS_CHECK_FREQUENCY = 1.hour
+    BATCH_SIZE = 1000.0
+    OPEN_TIMEOUT = 2
+    REQUEST_TIMEOUT = 6
+    Pointer = Struct.new(:x, :y, :facility)
+
+    def initialize(url, _type, adapter)
+      @url = url
+      @adapter = adapter
+      @mutex = Mutex.new
+      @last_check = Time.new(0).utc
+      @current_gis_update = 0
+      @client = BulkClient.new(@url)
+      @index = {}
+      @coordinates = []
+      # check_for_freshness
+    end
+
+    def get(id)
+      check_for_freshness
+      @mutex.synchronize do
+        return @index[id]
+      end
+    end
+
+    def query(bbox, services = nil)
+      check_for_freshness
+      filter = services ? @adapter.with_services(services) : ->(_) { true }
```

Comment:
The models for the three different facility types (VHA/VBA/NCA) have differing representations of services - VHA has a hierarchical model, VBA has a flat list, NCA has no services. But there was a decision made to have them represented by the same model. So, the adapter classes provide a consistent approach for filtering on services regardless of what representation is used. I agree it can be done without a lambda, just a method on the adapter class, but I think given these constraints encapsulating the with_services logic in the adapter class is correct. 

And I can do without the ternary operator, it is just there to decide whether any services filtering was needed at all, for terseness of the chained methods below. Will de-terse that.